www/source/blog/2017-10-15-Configuration-Plans.html.md

```diff
@@ -0,0 +1,119 @@
+---
+title: Configuration Plans
+date: 2017-10-15
+author: Thom May
+tags: [plans, configuration, glossary]
+category: Packaging
+classes: body-article
+---
+
+One of the hardest problems in computer science is naming things. But one of the
+most important things for a healthy community is shared language. As the
+Habitat community grows, we'll try to identify patterns and practices
+that help us succeed, and give them names so that we can all talk about
+them.
+
+One pattern we use a lot is what we're calling a Configuration Plan. As
+we share our packages, we've been seeing that our colleagues and
+community members use those packages in ways that we could not have
+anticipated. So what to do? We could try and add templates to
+support all possible uses, but that will rapidly become untenable. What
+good packages should do instead is provide well documented entry points, so
+that consumers of our package can easily supply their own configuration
+in a new plan -- a Configuration Plan.
+
+## Building a Configuration Plan
+
+Let's have a look at an example. There are actually a fair number of
+live examples in the habitat repository - check out the [Builder API
+Proxy](https://github.com/habitat-sh/habitat/tree/master/components/builder-api-proxy), which is a Configuration Plan for nginx. But for now, let's build our own.
+
+I've created a tiny binary that simply reads a file and prints it out.
+You can [see the
+source](https://github.com/thommay/configuration_plan_demo), and you can
```

Comment:
In the realm of naming hard things: would you consider a different name for this package?  Something that perhaps denotes that it's the application with with the default/standard configuration.

Reply:
Or call it what it does: config file reader/reporter?